packages/gestalt/src/behaviors/OutsideEventBehavior.js

```diff
@@ -1,39 +1,35 @@
 // @flow
-import * as React from 'react';
-import { findDOMNode } from 'react-dom';
+import type { Element } from 'react';
+import React, { useEffect } from 'react';
 
 type Props = {|
-  children: React.Node,
+  children: Element<*> | Array<Element<*>>,
   onClick?: (event: MouseEvent) => void,
 |};
 
-export default class OutsideEventBehavior extends React.Component<Props> {
-  componentDidMount() {
-    document.addEventListener('click', this.handleClickEvent, {
-      capture: true,
-    });
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('click', this.handleClickEvent, {
-      capture: true,
-    });
-  }
-
-  handleClickEvent = (event: MouseEvent) => {
-    // eslint-disable-next-line react/no-find-dom-node
-    const el = findDOMNode(this);
-    if (
-      !this.props.onClick ||
-      !el ||
-      (event.target instanceof Node && el.contains(event.target))
-    ) {
+const OutsideEventBehavior = ({ children, onClick }: Props) => {
+  let isClickedInside = false;
```

Comment:
@emilyalwayss I'm curious to see if we can use React `useState` for this variable instead. Could you give that a try?

Reply:
@christianvuerings What are some of the benefits of storing it as a state variable? I don't think we want to trigger a re-render when setting this variable.

Storing as a state var also introduces some unexpected behavior (ie. clicking outside of the modal only clears the flyout). However, if you feel we should do it this way, I will look into it!